### PR TITLE
Add allowed principals to service endpoints

### PR DESCRIPTION
**Pull request by csrwng**
**What this PR does / why we need it**:
Before this commit creating a private link between a management cluster
and a hosted cluster in a different AWS account failed because the
service endpoints created on the management side did not allow endpoints
on the guest side to connect.

This PR adds the control plane role as an allowed principal to the
endpoint service, making it possible to establish the private link
across accounts.

The way it obtains the control plane role is by reading the control
plane credentials secret referenced by the hosted cluster. In order
to parse the secret (in ini format), it vendors the
gopkg.in/ini.v1 package.

**Which issue(s) this PR fixes**:
Fixes # [HOSTEDCP-408](https://issues.redhat.com/browse/HOSTEDCP-408)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: hypershift-operator/controllers/platform/aws/controller.go [hunk 1]
```diff
@@ -343,76 +349,116 @@ func reconcileAWSEndpointServiceStatus(ctx context.Context, c client.Client, aws
...
+			log.Info("endpoint service created", "serviceName", serviceName)
 		}
-		return err
 	}
-	if len(output.LoadBalancers) == 0 {
-		return fmt.Errorf("load balancer %s not found", lbName)
+	awsEndpointService.Status.EndpointServiceName = serviceName
+
+	// reconcile permissions for aws endpoint service
+	if hostedCluster == nil {
```
[the remark this answers is not in the document]
Comment (csrwng) [reply hypershift-operator/controllers/platform/aws/controller.go:425]: I had put this here so the unit test would skip the rest of the function. However, I think the better thing to do is to test the rest of the function in the unit test as well.

File: hypershift-operator/controllers/platform/aws/controller.go [hunk 2]
```diff
@@ -184,16 +188,27 @@ func (r *AWSEndpointServiceReconciler) Reconcile(ctx context.Context, req ctrl.R
...
+	// Find the hosted control plane
+	hcp, err := r.hostedControlPlane(ctx, awsEndpointService.Namespace)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
 	// Reconcile the AWSEndpointService Spec
 	if _, err := r.CreateOrUpdate(ctx, r.Client, awsEndpointService, func() error {
-		return reconcileAWSEndpointServiceSpec(ctx, r, awsEndpointService)
+		return reconcileAWSEndpointServiceSpec(ctx, r, awsEndpointService, hcp)
```
[the remark this answers is not in the document]
Comment (csrwng) [reply hypershift-operator/controllers/platform/aws/controller.go:199]: Yup, makes sense

**Comment by csrwng** [comment]
@enxebre I addressed your comments

**Comment by sjenning** [comment]
Wish there was a cleaner way to do this :sweat_smile: but is what it is for now

Thanks @csrwng!

/lgtm

**Pull request merged**